### PR TITLE
Delivery costs with clear history

### PR DESCRIPTION
**Pull request by mileo**

### CODE REVIEW

**Comment by mileo** [comment]
@lfdivino Bom dia! Manda os testes que vc fez do frete nessa branch, pra subir o coverage!

**Comment by mileo** [comment]
https://travis-ci.org/kmee/l10n-brazil/builds/91436412

**Comment by rvalyi** [comment]
@mileo you need to ensure @lfdivino signed the OCA CLA and that he enabled in his Github account the mail address he used to sign the CLA. After that push again with -f and the CLA issue should be void.

**Comment by mileo** [comment]
@lfdivino did u singed?

**Comment by mileo** [comment]
@lfdivino  Push again with -f and the CLA issue should be void.

**Comment by mileo** [comment]
:-1: @lfdivino 
Please fix freight tests:
1 - Use fields of sale.order.line instead o sale.order like https://github.com/kmee/l10n-brazil/blob/clear-delivery-costs/l10n_br_account_product/models/account_invoice.py#L758
2 - Create a case test that use the button/action action_l10n_br_account_product_costs_ratio increasing the coverage

**Comment by lfdivino** [comment]
Okay @mileo i will fix it.

File: l10n_br_account_product/models/account_invoice.py [hunk 1]
```diff
@@ -492,6 +492,45 @@ def action_date_assign(self):
...
         return True
 
+    @api.multi
+    def button_reset_taxes(self):
+        result = super(AccountInvoice, self).button_reset_taxes()
+        ait = self.env['account.invoice.tax']
+        for invoice in self:
+            invoice.read()
+            costs = []
+            company = invoice.company_id
```
Comment (mileo) [inline l10n_br_account_product/models/account_invoice.py:498]: @lfdivino Please raise an excetion UserError if the tax are not filled in company form

**Comment by mileo** [comment]
@lfdivino 
1 - You have to cherry-pick this commit https://github.com/odoo-brazil/l10n-brazil/pull/30/files
2 - Add this accounts in demo data company form

**Comment by mileo** [comment]
@lfdivino any update?

**Comment by mileo** [comment]
@rvalyi Finished for now. New features will be add in a new PR.

**Comment by rvalyi** [comment]
@mileo cool thanks. @renatonlima and me got busy like mad this week with a consultancy in a factory in Saquarema, but hopefully we free some time for review soon, Would help too if we have reviews from @danimaribeiro too eventually. The good news is we will probably make some v7->v8 openupgrade scripts for the project and fix the NT thing for the 7.0 guys this way.

**Comment by renatonlima** [comment]
@mileo @lfdivino 

Can you rebase this branch and fix pep8?

**Comment by mileo** [comment]
@renatonlima FIXED, tests ok!
@rvalyi

**Pull request merged**